processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -426,4 +518,37 @@ public Row apply(Object input)
       }
     };
   }
+
+
+  public static Set<String> optimizedExtractionDims(GroupByQuery query, final boolean getPre)
+  {
+    return Sets.newHashSet(
```

Comment:
A part of me thinks that a for loop would actually be more concise and easier to read...